packages/lesswrong/components/editor/EditorFormComponent.tsx

```diff
@@ -425,6 +425,7 @@ class EditorFormComponent extends Component<EditorFormComponentProps,EditorFormC
   resetEditor = () => {
     const { name, document } = this.props;
     // On Form submit, create a new empty editable
+    this.setEditorValue("");
```

Comment:
I'm a little confused as to how the bug was happening - is it because of `saveBackup()`? Also not sure if it's worth our time making sure I understand the bug.

My main concern is that I actually wrote the `setEditorValue()` function recently as kind of a hack to get my EAG import form working, so I would be worried about relying on it for anything else. Maybe all you need from that function is the call to `this.state.ckEditorReference?.setData()`?

Reply:
Yep - the problem occurs inside `saveBackup`. In that function, we check to see if the current editor state is empty; if it's empty we clear the local storage, otherwise we set local storage to be the editor state. The problem is that we don't check the editor state stored inside `EditorFormComponent`, we check the state that is stored internally inside the editor itself. This is the correct behaviour since the editor should always be the absolute source of truth whilst editing, but in this specific situation the component should be the source of truth because only the component is aware that we're submitting and clearing a form, and the internal editor state still contains the data from before the form is submitted.

Originally, I did call `this.state.ckEditorReference?.setData()` directly (it's in one of the commits I force-pushed over before submitting the PR), but I thought it'd be better to have this abstracted. Looking at it again fresh, I think you're right - this is just unnecessary. I'll change it back.

RE the tests - the cypress test have been totally broken on master for a couple of days. I meant to look at it this week but I didn't have time. Looking at the git history, it looks like a regression from [preloadDuringSSR](https://github.com/ForumMagnum/ForumMagnum/pull/5310), but that could be a red herring.

Reply:
Aah - I edited it on a different computer to usual and Vim got a bit trigger-happy with the white-space. I'm rolling with it...